File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.cs
public class Solution {
    public int NumSub(string s) {
        const int MOD = 1000000007;
        long ans = 0, count = 0;

        foreach (char c in s) {
            if (c == '1') {
                count++;
            } else {
                ans += (count * (count + 1) / 2) % MOD;
                count = 0;
            }
        }
        ans += (count * (count + 1) / 2) % MOD;

        return (int)ans;
    }
}

File: 1636-number-of-substrings-with-only-1s/number-of-substrings-with-only-1s.py
class Solution:
    def numSub(self, s: str) -> int:
        n = len(s)
        mod = 10**9 + 7
        ans, count = 0, 0

        for i in range(n):
            if s[i] == "1":
                count += 1
            else:
                ans += ((count * (count + 1)) // 2) % mod
                count = 0

        ans += ((count * (count + 1)) // 2)

        return ans
